### PR TITLE
[Tests-Only] Add more files related test

### PR DESCRIPTION
**Pull request by dpakach**
Fixes https://github.com/owncloud/owncloud-sdk/issues/756

Add more tests to cover files, trashbin, and favorites scenarios

### CODE REVIEW

File: tests/fileTrashTest.js [hunk 1]
```diff
@@ -973,5 +974,61 @@ describe('oc.fileTrash', function () {
...
+            resourcePath: path.join(testFolder, testFile),
+            username: testUser,
+            password: testUserPassword
+          })
+          .uponReceiving(`as '${testUser}', a MOVE request to restore file from trashbin to a non existing location`)
+          .withRequest({
+            method: 'MOVE',
+            path: MatchersV3.fromProviderState(
+              `/remote.php/dav/trash-bin/${testUser}/\${trashId}`,
+              '/remote.php/dav/trash-bin/' + testUser + '/' + deletedFolderId
```
Comment (saw-jan) [inline tests/fileTrashTest.js:1002]: yo can also use string literal here

File: tests/fileTrashTest.js [hunk 2]
```diff
@@ -973,5 +974,61 @@ describe('oc.fileTrash', function () {
...
+            }
+          })
+          .willRespondWith({
+            status: 409,
+            headers: applicationXmlResponseHeaders,
+            body: webdavExceptionResponseBody('Conflict', 'The destination node is not found')
+          })
+      }
+
+      it.only('Trying to restore file should give error', async function () {
```
Comment (saw-jan) [inline tests/fileTrashTest.js:1016]:
```suggestion
      it('Trying to restore file should give error', async function () {
```

**Review by saw-jan — COMMENTED** [review]
LGTM :+1:

File: tests/filesTest.js [hunk 3]
```diff
@@ -1351,6 +1374,61 @@ describe('Main: Currently testing files management,', function () {
...
+    it('checking method getFavoriteFiles when there are no favorite files', async () => {
+      const provider = createProvider(false, true)
+      await getCapabilitiesInteraction(provider, config.testUser, config.testUserPassword)
+      await getCurrentUserInformationInteraction(
+        provider, config.testUser, config.testUserPassword
+      )
+      await givenUserExists(provider, config.testUser, config.testUserPassword)
+      await givenFileExists(provider, config.testUser, config.testUserPassword, file)
+      await provider
+        .uponReceiving(`as '${username}', a REPORT request to get favorite file when there are not any favorites`)
```
Comment (individual-it) [inline tests/filesTest.js:1388]:
```suggestion
        .uponReceiving(`as '${username}', a REPORT request to get favorite file when there are no favorites`)
```

**Review by individual-it — CHANGES_REQUESTED** [review]

**Review by individual-it — APPROVED** [review]

**Pull request merged**